packages/mobile/src/geth/saga.ts

```diff
@@ -98,6 +101,74 @@ export function* waitForGethConnectivity() {
   }
 }
 
+function* pollGethStatusUntilSynced() {
+  let head: BlockHeader = yield select(chainHeadSelector)
+
+  // Head may be null on startup
+  if (!head) {
+    head = (yield take(Actions.SET_CHAIN_HEAD)).head
+  }
+
+  while (true) {
+    if (blockIsFresh(head)) {
+      return true
+    }
+
+    head = (yield take(Actions.SET_CHAIN_HEAD)).head
+  }
+}
+
+export function* waitForGethSync() {
+  const fornoEnabled = yield select(fornoSelector)
+  if (fornoEnabled) {
+    return
+  }
+
+  const { result } = yield race({
+    result: call(pollGethStatusUntilSynced),
+    timeout: delay(GETH_SYNC_TIMEOUT),
+  })
+
+  if (!result) {
+    throw Error(`Geth failed to sync within ${GETH_SYNC_TIMEOUT / 1000} seconds`)
+  }
+}
+
+function pollGethSyncStatusAsync() {
+  return new Promise(async (resolve, reject) => {
+    const dateLimit = Date.now() + GETH_SYNC_TIMEOUT
+
+    while (Date.now() <= dateLimit) {
```

Comment:
This polling pattern should be generic and created as a util function if not already